File: frontend/types/api/projects.ts
export interface ProjectFile {
  id: string
  filename: string
  file_type: string | null
  storage_path: string
  created_at: string
}

export interface Project {
  id: string
  owner_id: string
  name: string
  description: string | null
  is_public: boolean
  status: string | null
  github_repo_url: string | null
  secret_link?: string | null
  created_at: string
  updated_at: string
  files: ProjectFile[]
}

export interface ProjectListResponse {
  items: Project[]
  total: number
  page: number
  size: number
}

export interface ProjectUploadResult {
  filename: string
  storage_path: string
}

export interface ProjectUploadResponse {
  project: Project
  upload_result: ProjectUploadResult | null
}

export interface PreviewAsset {
  id: string
  filename: string
  title?: string
  page?: number
  page_count?: number
  pages?: PreviewAsset[]
  multi_page?: boolean
  sources?: string[]
  composed?: PreviewAsset
  layers?: string[]
  placeholder?: boolean
  path?: string
  url?: string | null
}

export interface ProjectPreviewResponse {
  project: Record<string, unknown>
  schematics: PreviewAsset[]
  layouts: PreviewAsset[]
  models: PreviewAsset[]
}

export interface ProjectCreatePayload {
  name: string
  description?: string | null
  is_public?: boolean
  github_repo_url?: string | null
}

export interface ProjectUpdatePayload {
  name?: string | null
  description?: string | null
  is_public?: boolean | null
  status?: string | null
  github_repo_url?: string | null
}

export interface ListProjectsQuery {
  page?: number
  size?: number
  only_public?: boolean
  owner_id?: string
}

File: backend/app/api/schemas/projects.py
"""Pydantic schemas for project management."""

from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class ProjectBase(BaseModel):
    name: str = Field(..., max_length=255)
    description: str | None = Field(default=None)
    is_public: bool = True
    status: str | None = Field(default=None, max_length=50)
    github_repo_url: str | None = Field(default=None, max_length=500)


class ProjectCreate(ProjectBase):
    owner_id: UUID | None = Field(default=None)


class ProjectUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=255)
    description: str | None = Field(default=None)
    is_public: bool | None = None
    status: str | None = Field(default=None, max_length=50)
    github_repo_url: str | None = Field(default=None, max_length=500)


class ProjectFileResponse(BaseModel):
    id: UUID
    filename: str
    file_type: str | None
    storage_path: str
    created_at: datetime


class ProjectResponse(ProjectBase):
    id: UUID
    owner_id: UUID
    processing_status: str
    processing_error: str | None = None
    view_count: int = 0
    open_comment_count: int = 0
    total_comment_count: int = 0
    created_at: datetime
    updated_at: datetime
    files: list[ProjectFileResponse] = Field(default_factory=list)

    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    items: list[ProjectResponse]
    total: int
    page: int
    size: int


class ProjectUploadResponse(BaseModel):
    project: ProjectResponse
    upload_result: dict[str, Any] | None = None


class ProjectPreviewResponse(BaseModel):
    project: dict[str, Any]
    schematics: list[dict[str, Any]]
    layouts: list[dict[str, Any]]
    models: list[dict[str, Any]]
